app/Http/Controllers/Frontend/BoardController.php

```diff
@@ -42,13 +42,17 @@ public function show(Request $request, Board $board)
         }
 
         $postsQuery->orderBy($orderBy, $request->sort === 'oldest' ? 'asc' : 'desc');
-        $posts = $postsQuery->get();
+        $posts = $postsQuery->cursorPaginate(20);
 
         $data = [
             'board' => $board,
             'posts' => $posts,
         ];
 
+        if ($request->cursor) {
```

Comment:
`$request->wantsJson()` is recommend for api response.

```suggestion
if ($request->wantsJson()) {
    return response()->json($data);
}
```